src/base/porting_inline.hpp

```diff
@@ -34,14 +34,17 @@
 #endif
 
 #if defined (__cplusplus) || defined (__GNUC__)
+#define EXTERN_INLINE extern inline
 #define STATIC_INLINE static inline
 #define INLINE inline
 #elif _MSC_VER >= 1000
+#define EXTERN_INLINE __forceinline extern
 #define STATIC_INLINE __forceinline static
 #define INLINE __forceinline
 #else
 /* TODO: we have several cases of using INLINE/STATIC_INLINE and adding function definition in headers. This won't
  * work. */
+#define EXTERN_INLINE extern
 #define STATIC_INLINE static
 #define INLINE
 #endif
```

Comment:
In other code, They only use STATIC_INLINE (dbtype_function.i) even though the functions are exposed. Could you explain why we need EXTERN_INLINE for this case? What is the difference between extern inline and static inline?

Reply:
I will use static inline consistently.

TMI.
when compiling with g++, there was no difference between static inline, inline and extern inline in assembly code.
However, When compiling with gcc, multiple definition errors occur when extern inline is used.
We are compiling our c files with g++, so no error occurs.